pallene/coder.lua

```diff
@@ -1118,6 +1123,64 @@ gen_cmd["SetArr"] = function(self, cmd, _func)
     }))
 end
 
+gen_cmd["NewTable"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dst)
+    local n = C.integer(cmd.size_hint)
+    return (util.render([[ $dst = pallene_new_table(L, $n); ]], {
+        dst = dst,
+        n = n,
+    }))
+end
+
+gen_cmd["GetTable"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dst)
+    local tab = self:c_value(cmd.src_tab)
+    local key = self:c_value(cmd.src_k)
+    local dst_typ = cmd.dst_typ
+    local line = C.integer(cmd.loc.line)
+
+    return (util.render([[
+        {
+            static int cache = 0;
+            TValue *slot = pallene_getstr($tab, $key, &cache);
+            ${check_tag}
+            $dst = $get_slot;
+        }
+    ]], {
+        dst = dst,
+        tab = tab,
+        key = key,
+        line = line,
+        check_tag = self:check_tag(dst_typ, "slot", cmd.loc, "table field"),
+        get_slot = get_slot(dst_typ, "slot"),
+    }))
+end
+
+gen_cmd["SetTable"] = function(self, cmd, _func)
+    local tab = self:c_value(cmd.src_tab)
+    local key = self:c_value(cmd.src_k)
+    local v = self:c_value(cmd.src_v)
+    local src_typ = cmd.src_typ
+    local line = C.integer(cmd.loc.line)
+    return (util.render([[
+        {
+            static int cache = 0;
+            TValue *slot = pallene_getstr($tab, $key, &cache);
+            if (PALLENE_UNLIKELY(isabstkey(slot))) {
```

Comment:
Well, it could be an error. Probably making it an error will bump the performance a little bit because the slow-path won't return. Can we go in this direction?

Reply:
I implemented it this way to be consistent with the behaviors of arrays, that allow you to insert keys in positions that are nil. Anyway, triggering an error is fine by me.

Reply:
If we decide to make it an error, we'd have to be careful about correctly dealing with fields that have type `value` or `nil`. In these cases, a hole in the table might not be an error.